mlrun/db/httpdb.py

```diff
@@ -312,9 +312,26 @@ def paginated_api_call(
         headers=None,
         timeout=45,
         version=None,
+        return_all=False,
     ) -> typing.Generator[requests.Response, None, None]:
         """
-        Calls the api with pagination, yielding each page of the response
+        Calls the API with pagination and yields each page of the response.
+
+        Depending on the `return_all` parameter:
+        - If `return_all` is `True`, fetches and yields all pages of results.
+        - If `return_all` is `False`, fetches and yields only the first page of results.
```

Comment:
it's more "a single page" rather than "the first page", because you can get a specific page != 1